assets/src/components/editing/elements/foreign/ForeignEditor.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useCallback } from 'react';
+import { useEditModelCallback } from 'components/editing/elements/utils';
+import * as ContentModel from 'data/content/model/elements/types';
+import { useElementSelected } from '../../../../data/content/utils';
+import { EditorProps } from 'components/editing/elements/interfaces';
+import { HoverContainer } from '../../toolbar/HoverContainer';
+import { Toolbar } from '../../toolbar/Toolbar';
+import { CommandButton } from '../../toolbar/buttons/CommandButton';
+import { createButtonCommandDesc } from '../commands/commandFactories';
+import { ForeignModal } from './ForeignModal';
+import { modalActions } from '../../../../actions/modal';
+
+interface Props extends EditorProps<ContentModel.Foreign> {}
+
+const dismiss = () => window.oliDispatch(modalActions.dismiss());
+const display = (c: any) => window.oliDispatch(modalActions.display(c));
+
+export const ForeignEditor = (props: Props) => {
+  const onEdit = useEditModelCallback(props.model);
+  const selected = useElementSelected();
+
+  const onModalDone = useCallback(
+    (changes: Partial<ContentModel.Foreign>) => {
+      onEdit({
+        ...props.model,
+        ...changes,
+      });
+      dismiss();
+    },
+    [onEdit, props.model],
+  );
+
+  const onChangeLang = useCallback(
+    () =>
+      display(
+        <ForeignModal
+          commandContext={props.commandContext}
+          model={props.model}
+          onDone={onModalDone}
+          onCancel={dismiss}
+        />,
+      ),
+    [onModalDone, props.commandContext, props.model],
+  );
+
+  return (
+    <HoverContainer
+      position="bottom"
+      align="start"
+      isOpen={selected}
+      content={
+        <Toolbar context={props.commandContext}>
+          <Toolbar.Group>
+            <CommandButton
+              description={createButtonCommandDesc({
+                icon: 'edit',
+                description: 'Change Language',
+                execute: onChangeLang,
+              })}
+            />
+          </Toolbar.Group>
+        </Toolbar>
+      }
+    >
+      <span className="foreign-editor" lang={props.model.lang} {...props.attributes}>
```

Comment:
Note: In the editor, we don't currently get the project-default setting for lang, but in the preview we do.

Reply:
I don't think we need it in the pure authoring context.